docs/source/api/notebooks/Notebook_4_Customized_Models.ipynb

```diff
@@ -0,0 +1,456 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Notebook 4: Use GraphStorm APIs for Customizing Model Components\n",
+    "\n",
+    "This notebook provides an example about how to customize components of GML models to fit specific requirements. These customized models normally extend GraphStorm higher-level APIs so that they not only implement their own functionalities, but also can embed into GraphStorm training and inference pipelines easily.\n",
+    "\n",
+    "----\n",
+    "\n",
+    "### Customized Model Example\n",
+    "\n",
+    "A widely used GNN model is the `RGAT` model, proposed by the [Relational Graph Attention Networks](https://arxiv.org/abs/1904.05811). The original `RGAT` model considers the different importants of neighbors for a node, and leverages attention mechanism to aggregate messages from neighbors within the same edge type. Then aggregations from neighbors cross different relation types are added together as the output representations of a node.\n",
+    "\n",
+    "$$h_i = \\sum_{r\\in \\mathscr{R}}\\sum_{j\\in \\mathcal{N}^{r}_{(i)}} \\alpha^{r}_{i,j} W^{r} h_j^{r}$$\n",
+    "\n",
+    "One way to enhance the `RGAT` model is to add attention mechnism to the aggregations **cross different relation types** instead of simply summing them up. We can use an additional weight set to compute the attention coefficients for different relation type, and then  compute the weighted sum of aggregations cross relation types.\n",
```

Comment:
```suggestion
    "One way to enhance the `RGAT` model is to add attention mechnism to the aggregations **cross different relation types** instead of simply summing them up. We can use an additional weight set to compute the attention coefficients for different relation types, and then  compute the weighted sum cross relation types.\n",
```

Reply:
I am not sure whether we should call it the enhancement of the 'RGAT'. It is not proved that attention on relations works better than sum.
We can just say An alternative way to aggregate messages from different relation types is using attention instead of summation.

Reply:
Good call. Updated.